### DISCUSSION TITLE
How to use the logging by referencing the logger provider

### DISCUSSION THREAD

**Discussion by akhansari** [aws-powertools/powertools-lambda-dotnet#545] [Q&A] [answered]
Hi,

How can we use the Powertools logging without static references, only by registering its logger provider and using the Microsoft.Extensions.Logging abstractions?

I've tried the following code but I'm not sure if the configuration setup is taken into account or not:

```csharp
using AWS.Lambda.Powertools.Logging.Internal;

...

builder.Logging.ClearProviders();
builder.Logging.AddProvider(new LoggerProvider(null));
```

**Comment by hjgraca** [comment, marked as answer]
Hi @akhansari ,

First of all, thanks for using Powertools!

It is not documented but there is a way you can use ILoggingBuilder, ServiceCollection and Dependency Injection with Powertools logger.
Have a look at this [gist](https://gist.github.com/hjgraca/f84ff9edf32f7ccc58ef038a0c93043f)

Hope this helps.
